File: frontend/src/microservices/tour-group/objects/GroupSelectionOrderProjectionResponse.ts
// 本文件定义 tour-group 模块的 `GroupSelectionOrderProjectionResponse`，作为响应数据并提供 JSON 编解码。

export type GroupSelectionOrderProjectionResponse = {
  selectionId: string
  orderId: string
  orderStatus: string
  paymentStatus: string
  supplierReviewStatus: string
  refundStatus: string | null
  bookingSummaryLabel: string
}
export const groupSelectionOrderProjectionResponseFromJson = (json: string): GroupSelectionOrderProjectionResponse =>
  JSON.parse(json) as GroupSelectionOrderProjectionResponse

export const groupSelectionOrderProjectionResponseToJson = (value: GroupSelectionOrderProjectionResponse): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/tour-group/objects/GroupSelectionOrderProjectionResponse.scala
package com.typesafe.travel.tourgroup.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class GroupSelectionOrderProjectionResponse(
    selectionId: String,
    orderId: String,
    orderStatus: String,
    paymentStatus: String,
    supplierReviewStatus: String,
    refundStatus: Option[String],
    bookingSummaryLabel: String
)

object GroupSelectionOrderProjectionResponse:
  given sourceEncoder: Encoder[GroupSelectionOrderProjectionResponse] = deriveEncoder
  given sourceDecoder: Decoder[GroupSelectionOrderProjectionResponse] = deriveDecoder
